### ISSUE TITLE
Update hyperparameters to be same as sklearn defaults

### ISSUE THREAD

**Issue by desilinguist** [EducationalTestingService/skll#438]
scikit-learn plans to change some values for the default model hyperparameters (e.g., `max_iter` and `tol` for `SGDClassifier` and `SGDRegressor`). We should figure out when and if we also want to update these when we release v2.0. 

Right now many of these lead to `FutureWarnings` in the current version and will continue to do so in the upcoming v1.5.3 release.

**Comment by desilinguist** [comment]
@aoifecahill @bwriordan The scikit-learn folks are [planning to change](https://github.com/scikit-learn/scikit-learn/issues/8361) the default value of `gamma` for `SVR` models from `auto` to `scaled` in v0.22 to better account for unscaled features. 

To account for this, I am thinking of changing the default value of `gamma` to `scaled` from `auto` and change the default parameter grid from:

```
['auto', 0.01, 0.1, 1.0, 10.0, 100.0]}]
```

to 

```
['auto', 'scale', 0.01, 0.1, 1.0, 10.0, 100.0]}]
```

This would have the benefit of providing a better default without grid search but still allow `auto` to be picked during grid search.

**Comment by desilinguist** [comment]
Another potentially result-changing change is that the `iid` parameter of `GridSearchCV`:

> If `True`, return the average score across folds, weighted by the number of samples in each test set. In this case, the data is assumed to be identically distributed across the folds, and the loss minimized is the total loss per sample, and not the mean loss across the folds. If `False`, return the average score across folds. Default is True, but will change to False in version 0.22, to correspond to the standard definition of cross-validation. 
> 
> Changed in version 0.20: Parameter iid will change from True to False by default in version 0.22, and will be removed in 0.24.

**Comment by mulhod** [comment]
I prefer 1 since it is the next major version, as you mention. Also, `iid` should be `False` by default, anyway. So, no matter what, it seems like a good change.

**Comment by bndgyawali** [comment]
I would also vote for 1.

**Comment by jbiggsets** [comment]
Yup, 1 makes the most sense to me, too.

**Comment by aoifecahill** [comment]
I agree that 1 is probably the best option given the major update to 2.0

**Comment by desilinguist** [comment]
Addressed by #533
